### DISCUSSION TITLE
Dumping NTDS warning message

### DISCUSSION THREAD

**Discussion by GeneralBison** [Pennyw0rth/NetExec#329] [Q&A] [answered]
Hi, I was wondering if anyone could tell me about the specific risks posed by dumping NTDS on a Windows Server 2019 Domain Controller as mentioned in this rather alarming warning message:
https://github.com/Pennyw0rth/NetExec/blob/7af90207a8c2151ecc2fe43100ac639bf6612374/cme/crackmapexec.py#L264

I've not been able to find anything specific to Windows Server 2019 crashing, and it wasn't present in prior versions of CME (which I've been able to use successfully).

Is this message warning about general NTDS dumping risks or is there a specific reason it's singling out Windows Server 2019?
I'll continue to research, was just curious as I've not seen this message before.

**Comment by mpgn** [comment, marked as answer]
Because if the ntds database is hugggggge, using impacket lib will crash the DC. Therefore a warning is shown to avoid crashing the DC if you have no idea if the dc is huge or not :)

To avoid the crash, you simply dump a specific user or use the module ntdsutils

More info on the crash https://github.com/fortra/impacket/issues/1436

**Reply by GeneralBison** [reply]
Thanks for the info, the warning wasn't present in the version of CME that I'd been using previously so it was a bit of a shock, good to know the source.
